File: frontend/src/services/adminService.ts
import apiClient from './apiClient';
import {
  AdminIstatistikDto,
  AdminKullaniciDto,
  AdminKullaniciGuncelleRequest,
  AdminKullaniciSayfaResponse,
  SistemParametresiDto,
  SistemParametresiGuncelleRequest,
  SistemParametresiOlusturRequest,
} from '@/types/admin.types';

interface ApiResponse<T> {
  success: boolean;
  message: string | null;
  data: T;
}

export const adminService = {
  // İstatistikler
  async istatistikleriGetir(): Promise<AdminIstatistikDto> {
    const res = await apiClient.get<ApiResponse<AdminIstatistikDto>>('/admin/istatistikler');
    return res.data.data;
  },

  // Kullanıcı yönetimi
  async kullanicilariGetir(params: {
    arama?: string;
    plan?: string;
    aktif?: boolean;
    sayfa?: number;
    boyut?: number;
    siralama?: string;
    yon?: string;
  }): Promise<AdminKullaniciSayfaResponse> {
    const res = await apiClient.get<ApiResponse<AdminKullaniciSayfaResponse>>('/admin/kullanicilar', {
      params: {
        arama: params.arama || undefined,
        plan: params.plan || undefined,
        aktif: params.aktif !== undefined ? params.aktif : undefined,
        sayfa: params.sayfa ?? 0,
        boyut: params.boyut ?? 20,
        siralama: params.siralama ?? 'createdAt',
        yon: params.yon ?? 'desc',
      },
    });
    return res.data.data;
  },

  async kullaniciGetir(id: number): Promise<AdminKullaniciDto> {
    const res = await apiClient.get<ApiResponse<AdminKullaniciDto>>(`/admin/kullanicilar/${id}`);
    return res.data.data;
  },

  async kullaniciGuncelle(id: number, request: AdminKullaniciGuncelleRequest): Promise<AdminKullaniciDto> {
    const res = await apiClient.put<ApiResponse<AdminKullaniciDto>>(`/admin/kullanicilar/${id}`, request);
    return res.data.data;
  },

  async kullaniciSil(id: number): Promise<void> {
    await apiClient.delete(`/admin/kullanicilar/${id}`);
  },

  // Sistem parametreleri
  async parametreleriGetir(): Promise<SistemParametresiDto[]> {
    const res = await apiClient.get<ApiResponse<SistemParametresiDto[]>>('/admin/sistem-parametreleri');
    return res.data.data;
  },

  async parametreOlustur(request: SistemParametresiOlusturRequest): Promise<SistemParametresiDto> {
    const res = await apiClient.post<ApiResponse<SistemParametresiDto>>('/admin/sistem-parametreleri', request);
    return res.data.data;
  },

  async parametreGuncelle(id: number, request: SistemParametresiGuncelleRequest): Promise<SistemParametresiDto> {
    const res = await apiClient.put<ApiResponse<SistemParametresiDto>>(`/admin/sistem-parametreleri/${id}`, request);
    return res.data.data;
  },

  async parametreVarsayilanaGetir(id: number): Promise<SistemParametresiDto> {
    const res = await apiClient.put<ApiResponse<SistemParametresiDto>>(`/admin/sistem-parametreleri/${id}/varsayilan`, {});
    return res.data.data;
  },

  async parametreSil(id: number): Promise<void> {
    await apiClient.delete(`/admin/sistem-parametreleri/${id}`);
  },
};

File: backend/src/main/java/com/scinar/giderlerim/service/AdminService.java
package com.scinar.giderlerim.service;

import com.scinar.giderlerim.dto.admin.*;
import com.scinar.giderlerim.dto.response.ApiResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface AdminService {

    ApiResponse<AdminIstatistikDto> getIstatistikler();

    ApiResponse<Page<AdminKullaniciListeDto>> kullanicilariListele(String arama, String plan, Boolean aktif, Pageable pageable);

    ApiResponse<AdminKullaniciListeDto> kullaniciGetir(Long id);

    ApiResponse<AdminKullaniciListeDto> kullaniciGuncelle(Long id, AdminKullaniciGuncelleRequest request);

    ApiResponse<Void> kullaniciSil(Long id);

    ApiResponse<List<SistemParametresiDto>> parametreleriListele();

    ApiResponse<SistemParametresiDto> parametreGuncelle(Long id, SistemParametresiGuncelleRequest request);

    ApiResponse<SistemParametresiDto> parametreOlustur(SistemParametresiOlusturRequest request);

    ApiResponse<Void> parametreSil(Long id);

    ApiResponse<SistemParametresiDto> parametreVarsayilanaGetir(Long id);
}
